src/QMCWaveFunctions/detail/CUDA/cuBLAS_LU.cu

```diff
@@ -0,0 +1,212 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2021 QMCPACK developers.
+//
+// File developed by: Peter Doak, doakpw@ornl.gov, Oak Ridge National Laboratory
+//
+// File created by: Peter Doak, doakpw@ornl.gov, Oak Ridge National Laboratory
+//////////////////////////////////////////////////////////////////////////////////////
+
+#include "cuBLAS_LU.hpp"
+#include "Platforms/CUDA/cudaError.h"
+#include "Platforms/CUDA/cuBLAS.hpp"
+
+namespace qmcplusplus
+{
+namespace cuBLAS_LU
+{
+
+__device__ cuDoubleComplex complexDetLog(const double lu_diag, const int n_index, const int* pivots)
+{
+  cuDoubleComplex log_value;
+  log_value.x = log(abs(lu_diag));
+  log_value.y = ((lu_diag < 0) != ((pivots[n_index] - 1) == n_index)) * M_PI;
+  return log_value;
+}
+
+__device__ cuDoubleComplex complexDetLog(const cuDoubleComplex lu_diag, const int n_index, const int* pivots)
+{
+  cuDoubleComplex diag;
+  double pivot_factor = 1 - 2 * ((pivots[n_index] - 1) == n_index);
```

Comment:
Also seems the opposite sign.
why not straightforward.
```
pivots[n_index] - 1 == n_index ? 1 : -1
```

Reply:
Better to not have divergence.  Perhaps the compiler will produce something similar but likely not.